Crash when taking a video with Nexus 5

#### Software and hardware versions 
Collect v1.4.16, Android v6.0.1, Nexus 5

#### Problem description
Crash when taking a video with Nexus 5.

#### Steps to reproduce the problem
1. Open a form with a video widget such as Widgets from the test server
1. Tap "Record Video"

#### Other information 
`/storage/emulated/0/DCIM/Camera/VID_20170317_022056.mp4` doesn't exist. The path should be `/storage/self/primary/DCIM/Camera/VID_20170317_022056.mp4`.

```
I MediaWidget: Inserting VIDEO returned uri = content://media/external/video/media/264
D AndroidRuntime: Shutting down VM
E AndroidRuntime: FATAL EXCEPTION: main
E AndroidRuntime: Process: org.odk.collect.android, PID: 3373
E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=4, result=-1, data=Intent { dat=file:/storage/emulated/0/DCIM/Camera/VID_20170317_022056.mp4 flg=0x1 }} to activity {org.odk.collect.android/org.odk.collect.android.activities.FormEntryActivity}: java.lang.IllegalArgumentException: Unknown URL file:/storage/emulated/0/DCIM/Camera/VID_20170317_022056.mp4
E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:3699)
E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:3742)
E AndroidRuntime: 	at android.app.ActivityThread.-wrap16(ActivityThread.java)
E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1393)
E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:148)
E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: Unknown URL file:/storage/emulated/0/DCIM/Camera/VID_20170317_022056.mp4
E AndroidRuntime: 	at android.content.ContentResolver.delete(ContentResolver.java:1323)
E AndroidRuntime: 	at org.odk.collect.android.activities.FormEntryActivity.onActivityResult(FormEntryActivity.java:708)
E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:6456)
E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:3695)
E AndroidRuntime: 	... 9 more
W ActivityManager:   Force finishing activity org.odk.collect.android/.activities.FormEntryActivity
```

--- a/collect_app/src/main/java/org/odk/collect/android/widgets/VideoWidget.java
+++ b/collect_app/src/main/java/org/odk/collect/android/widgets/VideoWidget.java
@@ -271,10 +271,6 @@ public class VideoWidget extends QuestionWidget implements IBinaryWidget {
 
     @Override
     public void setBinaryData(Object binaryuri) {
-        // you are replacing an answer. remove the media.
-        if (mBinaryName != null) {
-            deleteMedia();
-        }
 
         // get the file path and create a copy in the instance folder
         String binaryPath = MediaUtils.getPathFromUri(this.getContext(), (Uri) binaryuri,
@@ -300,6 +296,10 @@ public class VideoWidget extends QuestionWidget implements IBinaryWidget {
             Log.i(t, "Inserting VIDEO returned uri = " + VideoURI.toString());
         } else {
             Log.e(t, "Inserting Video file FAILED");
+        }
+        // you are replacing an answer. remove the media.
+        if (mBinaryName != null && !mBinaryName.equals(newVideo.getName())) {
+            deleteMedia();
         }
 
         mBinaryName = newVideo.getName();

